src/libcrun/ebpf.c

```diff
@@ -293,15 +293,90 @@ bpf_program_complete_dev (struct bpf_program *program, libcrun_error_t *err arg_
   return program;
 }
 
+static int
+ebpf_attach_program (int fd, int dirfd, libcrun_error_t *err)
+{
+  const int MAX_ATTEMPTS = 20;
+  int attempt;
+
+  for (attempt = 0;; attempt++)
```

Comment:
maybe `attempt = 0; attempt < MAX_ATTEMPTS; attempt++` for readability

Reply:
I tried to rewrite using the condition inside of the for loop but there is some extra logic that decides whether to retry or not.